src/containers/Application/ApplicationOverview.tsx

```diff
@@ -118,3 +160,23 @@ const showProcessingModal = (processing: boolean, submitted: boolean) => {
 }
 
 export default ApplicationOverview
+
+function getFirstErrorLocation(
+  validityFailures: ValidityFailure[],
+  elementsState: ApplicationElementStates
+) {
+  let firstErrorSectionIndex = Infinity
+  let firstErrorPage = Infinity
+  validityFailures.forEach((failure: ValidityFailure) => {
+    const { code } = failure
+    const sectionIndex = elementsState[code].sectionIndex
+    const page = elementsState[code].page
+    if (sectionIndex < firstErrorSectionIndex) {
+      firstErrorSectionIndex = sectionIndex
+      firstErrorPage = page
+    } else
+      firstErrorPage =
+        sectionIndex === firstErrorSectionIndex && page < firstErrorPage ? page : firstErrorPage
+  })
+  return { firstErrorSectionIndex, firstErrorPage }
+}
```

Comment:
Very cool! I think this may be better included in the same file as `validateAll` so it's accessible for other client classes to use.

Reply:
Sure thing. Done 5d3ebaa

Reply:
I think I will probably use it again for #135 👍